geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/RedisSetInRegion.java

```diff
@@ -0,0 +1,115 @@
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.redis.internal.executor.set;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.regex.Pattern;
+
+import org.apache.geode.cache.Region;
+import org.apache.geode.redis.internal.ByteArrayWrapper;
+
+/**
+ * This class checks whether a RedisSet exists before forwarding the message.
+ */
+public class RedisSetInRegion implements RedisSetCommands {
+  private Region<ByteArrayWrapper, RedisSet> region;
+
+  @SuppressWarnings("unchecked")
+  public RedisSetInRegion(Region<ByteArrayWrapper, RedisSet> region) {
+    this.region = region;
+  }
+
+  @Override
+  public long sadd(
+      ByteArrayWrapper key,
+      ArrayList<ByteArrayWrapper> membersToAdd) {
+    RedisSet redisSet = get(key);
+    long l = redisSet.sadd(key, membersToAdd);
+    if (l > 0) {
+      region.put(key, redisSet);
+    }
+    return l;
+  }
+
+  @Override
+  public long srem(
+      ByteArrayWrapper key,
+      ArrayList<ByteArrayWrapper> membersToRemove, AtomicBoolean setWasDeleted) {
+    RedisSet redisSet = get(key);
+    long l = redisSet.srem(key, membersToRemove, setWasDeleted);
+    if (get(key).scard(key) == 0) {
+      region.remove(key);
+    } else {
+      region.put(key, get(key));
+    }
+    return l;
+  }
+
+  public boolean del(ByteArrayWrapper key) {
+    return region.remove(key) != null;
+  }
+
+  @Override
+  public Set<ByteArrayWrapper> smembers(
+      ByteArrayWrapper key) {
+    return get(key).smembers(key);
+  }
+
+  @Override
+  public int scard(ByteArrayWrapper key) {
+    return get(key).scard(key);
+  }
+
+  @Override
+  public boolean sismember(
+      ByteArrayWrapper key, ByteArrayWrapper member) {
+    return get(key).sismember(key, member);
+  }
+
+  @Override
+  public Collection<ByteArrayWrapper> srandmember(
+      ByteArrayWrapper key, int count) {
+    return get(key).srandmember(key, count);
+  }
+
+  @Override
+  public Collection<ByteArrayWrapper> spop(
+      ByteArrayWrapper key, int popCount) {
+    RedisSet redisSet = get(key);
+    Collection<ByteArrayWrapper> byteArrayWrappers = redisSet.spop(key, popCount);
+    if (get(key).scard(key) == 0) {
+      region.remove(key);
+    } else {
+      region.put(key, get(key));
+    }
+    return byteArrayWrappers;
+  }
+
+  @Override
+  public List<Object> sscan(
+      ByteArrayWrapper key, Pattern matchPattern, int count, int cursor) {
+    return get(key).sscan(key, matchPattern, count, cursor);
+  }
+
+  private RedisSet get(ByteArrayWrapper key) {
+    RedisSet redisSet = region.get(key);
+    return redisSet == null ? new RedisSet(new ArrayList<>()) : redisSet;
```

Comment:
I prefer getting null back from this method to prevent garbage creation. I know you prefer keeping the callers simpler by always returning a non-null. Can you return a canonical empty instance instead of always creating a new  RedisSet  and a new ArrayList?